src/Umbraco.Core/Services/IIconService.cs

```diff
@@ -16,6 +16,6 @@ public interface IIconService
         /// Gets a list of all svg icons found at at the global icons path.
         /// </summary>
         /// <returns></returns>
-        IList<IconModel> GetAllIcons();
+        IReadOnlyDictionary<string, string> GetAllIcons();
```

Comment:
This should be a new method on a v2 interface, the previous method should be left in place
        IReadOnlyDictionary<string, string> GetAllIconsDictionary();

Reply:
Yep, we cannot change signatures on existing methods for service interfaces. We can only change service interfaces by being additive. 

> the previous method should be left in place

And marked as [Obsolete] and with [EditorBrowsable(EditorBrowsableState.Never)]

If you are going to keep this same name GetAllIcons then unfortunately even on an `IIconService2` interface you will need to explicitly implement that method.

Welcome to the wonderful world of breaking changes :) In the future, almost 100% of the time, an interface should not have a mutable collection returned from a method/property.

Reply:
Guess that then raises the question do we want to cache both the old List<IconModel> and new IReadyOnlyDictionary<string, string> objects, even though the former is obsolete, not used internally, and (IMO) highly unlikely to have third-party consumers.

Completely understand the interface being additive.

Reply:
Just cache what we are using, nobody will be using this API and if for some crazy reason they are I think they can be persuaded to change.

Reply:
Works out ok - cache the dictionary, and cast that to the list if it's ever requested (which it won't, surely)